addons/web/static/src/public/caps_lock_warning_widget.js

```diff
@@ -0,0 +1,28 @@
+import PublicWidget from "@web/legacy/js/public/public_widget";
+
+export const CapsLockWarningWidget = PublicWidget.Widget.extend({
+    selector: ".oe_login_form, .oe_reset_password_form",
+    events: {
+        'keydown #password': '_onKeyDownEvent',
```

Comment:
I researched a bit on what debounce does and how does it work. I realized it is something that is fruitful to us in case where we wish to wait for the user to finish typing and then wish to trigger our function.

Let's say that we want to show suggestions for a search query, but only after a visitor has finished typing it.

But that is not the case here. In our case, we wish to process each and every input and toggle the warning on every subsequent input.

Just for the reference, here is the link to the article that I referred --- https://www.freecodecamp.org/news/javascript-debounce-example/